### ISSUE TITLE
A11y_Azure Data Studio extension 'Azure SQL Migration'_Databases for assessment_ScreenReader: Narrator/NVDA is announcing number of rows incorrectly, doesnt announce row name and row number is read incorrectly

### ISSUE THREAD

**Issue by NiharikaTipirishetty** [microsoft/azuredatastudio#19864]
"[Check out Accessibility Insights! ](https://accessibilityinsights.io/)- Identify accessibility bugs before check-in and make bug fixing faster and easier.”

### Github Tags:
#A11yMAS; #A11yTCS; #Benchmark; #SH-AzureSQLMigration-Win32-Jun2022; #DesktopApp; #Win32; #SQL Migration Control Plane; #WCAG1.3.1; #ScreenReader; #Narrator; #NVDA; #A11ySev3; #External; #External:NVDA; #ThirdParty;

### Environment and OS details:
Azure Data Studio Version 1.37.0(User Setup)
Windows Version 21H2 (OS Build 22000.675)

### Repro Steps:

1. Turn on Narrator/NVDA
2. Install Azure data studio app and install Azure SQL migration extension
3. Connect to SQL server, tab till respective server, right click and select manage
4. Tab till Azure SQL Migration in sitemap and press enter key
5. Tab till migrate to azure sql and press enter
6. Navigate to databases grid in "Databases for assessment" step.
7. Verify that screen reader announces table summary incorrectly and doesnot announce row name

### Actual Result:
Narrator/NVDA is announcing number of rows incorrectly, doesnt announce row name and row number is read incorrectly

### Expected Result:
Narrator/NVDA should announce total number of rows correctly, should announce row name and all the table properties must be defined.

### Attachment:
![Narrator_table summary_](https://user-images.githubusercontent.com/83570638/176399074-be56728f-77d4-44d4-8f76-b1e90b43b82d.gif)
![NVDA_table summary](https://user-images.githubusercontent.com/83570638/176401035-baaa89d5-2776-42c8-9bf0-5beb89a0f174.png)

**Comment by NiharikaTipirishetty** [comment]
Issue is not fixed. There are total 112 records present in the grid but screen reader is announcing table summary as 22/6 and when focus is on 2nd row, narrator is announcing as row 15 which is incorrect and while navigating using ctrl+alt+arrow keys, column header name for particular cell is not announced. 

Please find attachments for reference

![19864_narrator_not-fixed](https://user-images.githubusercontent.com/83570638/192492083-6af434f4-7744-4a63-b928-58b9ab3612e0.gif)
![19864_nvda_not fixed](https://user-images.githubusercontent.com/83570638/192493007-123135fd-472a-4f22-9c10-15fc905446d7.png)

**Comment by aasimkhan30** [comment]
@alanrenmsft , we are doing everything correctly on the ADS side. This seems to be an screen reader issue: https://github.com/nvaccess/nvda/issues/14332

**Comment by aasimkhan30** [comment]
@alanrenmsft, I am guessing we can open a similar bug for narrator. What do you suggest here?

**Comment by alanrenmsft** [comment]
@aasimkhan30  have you ruled out the possibility of this being an Electron issue?

**Comment by aasimkhan30** [comment]
Yeah, there is an example attached to the nvda bug. It happens on chrome as well. So, I don't think this is an electron issue.

**Comment by alanrenmsft** [comment]
I see, yes, please open issue for the screen readers.

**Comment by alanrenmsft** [comment]
the issue should be closed in this case, right? @NiharikaTipirishetty

**Comment by brian-harris** [comment]
@alanrenmsft , @NiharikaTipirishetty can we close this bug yet, it is giving us a "D" grade instead of "C".  Please close if possible

**Comment by alanrenmsft** [comment]
@brian-harris please reach out to the testing team directly.

**Comment by VijayaBharathyA** [comment]
We cannot close this bug since the issue is not fixed

**Comment by aasimkhan30** [comment]
Here is an example of this issue: https://codepen.io/aasimkhan30/pen/vYzLVYR

Can you please create issues for the screen readers?

**Comment by brian-harris** [comment]
@VijayaBharathyA can we please close this bug now and open an 3rd party issue.  This bug is giving SQL-Migration extension a D grade for an external issue.  The sample has been provided..

**Comment by brian-harris** [comment]
@VijayaBharathyA This bug seems to be related to a screen reader issue and not the ADS azdata.TableComponent component or the SQL-Migration extension. I am closing this bug.

Please open a 3rd party issue with the screen readers.   Link to example:  : https://codepen.io/aasimkhan30/pen/vYzLVYR

This is a sample of how to repro/create the azdata.TableComponent for reference as well:
private _databaseSelectorTable!: azdata.TableComponent;
		this._databaseSelectorTable = this._view.modelBuilder.table()
			.withProps({
				data: [],
				width: 650,
				height: '100%',
				forceFitColumns: azdata.ColumnSizingMode.ForceFit,
				columns: [
					<azdata.CheckboxColumn>{
						value: '',
						width: 10,
						type: azdata.ColumnType.checkBox,
						action: azdata.ActionOnCellCheckboxCheck.selectRow,
						resizable: false,
						cssClass: cssClass,
						headerCssClass: cssClass,
					},
					{
						value: 'databaseicon',
						name: '',
						width: 10,
						type: azdata.ColumnType.icon,
						headerCssClass: cssClass,
						cssClass: cssClass,
						resizable: false,
					},
					{
						name: constants.DATABASE,
						value: 'database',
						type: azdata.ColumnType.text,
						width: 360,
						cssClass: cssClass,
						headerCssClass: cssClass,
					},
					{
						name: constants.STATUS,
						value: 'status',
						type: azdata.ColumnType.text,
						width: 80,
						cssClass: cssClass,
						headerCssClass: cssClass,
					},
					{
						name: constants.SIZE,
						value: 'size',
						type: azdata.ColumnType.text,
						width: 80,
						cssClass: cssClass,
						headerCssClass: cssClass,
					},
				]
			}).component();

**Comment by VijayaBharathyA** [comment]
#External; #External:NVDA; #ThirdParty; #eDAD:3P:206523;
@brian-harris 
External bug Logged For NVDA : https://dev.azure.com/CSS-IRT/CSS%20IRT/_workitems/edit/206523
Could you please Reopen the bug this bug can be closed once external bug is fixed and closed

**Comment by brian-harris** [comment]
@Charles-Gagnon any help on this issue? Unfortunately the SQL-Migration extension is still getting a "D" grade because of this bug in NVIDEA.  How is it possible for any extension to get better than a D grade if this grid is used?

**Comment by Charles-Gagnon** [comment]
@alanrenmsft For all a11y issues

**Comment by alanrenmsft** [comment]
@brian-harris here is the link to the 3rd party external bug process: https://microsoft.sharepoint.com/teams/cea11y/SitePages/External-Bugs-Process---Third-Party-Dependencies.aspx

in short: the bug has to be kept open.

**Comment by sohelmulani1998** [comment]
Downgrading to Sev3 since its issue with NVDA. #A11ySev3;

**Comment by mstechie** [comment]
GitHubTags: A11ySev3;#External;#External:NVDA;#ThirdParty;

**Comment by brian-harris** [comment]
@mstechie, s360 still seems to be picking up the Sev2 tag and still show this as an out of Sla Sev 2 issue.

**Comment by brian-harris** [comment]
Updated github tags to lower severity from 3 to 2.
removing-A11ySev2 
adding-A11ySev3,External,External:NVDA,ThirdParty;

**Comment by BhanuPrakashGit** [comment]
Hi @alanrenmsft tried to reverify the issue in the environment:
Version: 1.48.0-insider (user setup)
Commit: 135b5e3e1e6f87ef00bd88bf6395cb8a1b599c0f
Date: 2023-12-07T06:32:18.548Z
VS Code: 1.82.0
Electron: 25.9.6
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Windows_NT x64 10.0.22621
But getting below error, could you please look into this. VPN is connected.

![image](https://github.com/microsoft/azuredatastudio/assets/104351566/4c2a5c84-a8c8-42b8-9b2d-5f010ac11853)

**Comment by alanrenmsft** [comment]
@brian-harris could you please provide a test server?

**Comment by brian-harris** [comment]
our test lab went offline over the weekend, looking into getting it back.

**Comment by BhanuPrakashGit** [comment]
#GithubTags: #Regressed12-12-23; #Closed;

Verified the issue in the environment: 
Version: 1.48.0-insider (user setup)
Commit: 135b5e3e1e6f87ef00bd88bf6395cb8a1b599c0f
Date: 2023-12-07T06:32:18.548Z
VS Code: 1.82.0
Electron: 25.9.6
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Windows_NT x64 10.0.22621

Issue is fixed and does not repro. Hence closing the bug.
PFA for reference. 

[19864_Fixed.zip](https://github.com/microsoft/azuredatastudio/files/13645130/19864_Fixed.zip)
